### ISSUE TITLE
Decrease display duration after NFC operation

### ISSUE THREAD

**Issue by dschuermann** [open-keychain/open-keychain#1734]
After https://github.com/open-keychain/open-keychain/pull/1705 , the display duration after NFC operations has increased. Maybe due to nordpol's usage of the NFC reader mode instead of foreground dispatch? Other ideas inside the pull request.

This should be fixed for 3.9!

**Comment by petter-fidesmo** [comment]
@dschuermann This is due to: https://github.com/fidesmo/nordpol/blob/master/android/src/main/java/nordpol/TagDispatcher.java#L207

This workaround can be disabled by setting `broadcomWorkaround` argument of the factory method to `false`. The broadcom chipset / driver combo in question is not that common in newer phones and most operations in OKC are probably fast enough to not be interrupted. It was a problem on Nexus 5, and it might still persist in the latest version. @Franzaine has a Nexus 5 and he could test the functionality if you make the change in a branch, but it should work as good/bad as before the switch to Nordpol.

**Comment by dschuermann** [comment]
damn tradeoffs ;) I am in favor of keeping the workarounds. @Valodim wants to try some things later today, maybe he can also test with a Nexus 5.

**Comment by Valodim** [comment]
Roughly counting, it seems to be five seconds of delay, which fits with the `DELAY_PRESENCE` constant from TagDispatcher. I don't think a five second delay for every crypto operation is acceptable at all :(

I'm not sure this is right though, the broadcomWorkaround argument is already set to false?:
https://github.com/open-keychain/open-keychain/blob/master/OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/base/BaseSecurityTokenNfcActivity.java#L202

**Comment by petter-fidesmo** [comment]
Seems to be set to true, since that will call the following factory https://github.com/fidesmo/nordpol/blob/master/android/src/main/java/nordpol/TagDispatcher.java#L100 which defaults it to true.

**Comment by petter-fidesmo** [comment]
A less "binary" solution could be to provide this as a integer "milliseconds" rather than boolean. I'll not have the time to this change right away, but I would be happy to merge a PR for such a solution.

**Comment by Valodim** [comment]
Yeah, I noticed only after I wrote that comment above. I set it to false, that solves the problem.

Are there a lot of devices which benefit from this workaround? Also, are the five seconds actually necessary for some operations?

@dschuermann Right now, I feel like we don't lose anything compared to the previous behavior if we disable the workaround, so in the interest of getting a release out soon I would go for that.

**Comment by petter-fidesmo** [comment]
So it's a bit tricky to keep track of all devices. It is related to this issue: https://code.google.com/p/android/issues/detail?id=58773

There have been some updates, and it might be so that newer versions of Android actually fixes this problem. So there might be a diminishing amount of devices which requires this workaround.

**Comment by petter-fidesmo** [comment]
Regarding the delay, several of the users of Nordpol (including Fidesmo) requires this delay for successfully being able to do heavier operations with the card. E.g. generating a public private key pair or deleting a big application can take quite some time.

**Comment by dschuermann** [comment]
@Valodim yup, let's postpone this and disable the workaround for this release. But keep the issue open please.

**Comment by Valodim** [comment]
aight
